test/async-hooks/test-promise.promise-before-init-hooks.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const initHooks = require('./init-hooks');
+const { checkInvocations } = require('./hook-checks');
+
+const p = (new Promise(common.mustCall(executor)));
+
+// init hooks after promise was created
+const hooks = initHooks();
+hooks._allowNoInit = true;
+hooks.enable();
+
+p.then(afterresolution);
+
+function executor(resolve, reject) {
+  resolve(5);
+}
+
+function afterresolution(val) {
+  assert.strictEqual(val, 5);
+  const as = hooks.activitiesOfTypes('PROMISE');
+  assert.strictEqual(as.length, 1, 'one activity');
+  checkInvocations(as[0], { init: 1, before: 1 },
+                   'after resolution child promise');
+  return val;
+}
+
+process.on('exit', onexit);
+function onexit() {
+  hooks.disable();
+  hooks.sanityCheck('PROMISE');
+
+  const as = hooks.activitiesOfTypes('PROMISE');
+  assert.strictEqual(as.length, 1, 'one activity');
+
+  const a0 = as[0];
+  assert.strictEqual(a0.type, 'PROMISE', 'promise request');
+  assert.strictEqual(typeof a0.uid, 'number', 'uid is a number');
+  assert.strictEqual(a0.triggerId, 5, 'use parent uid as triggerId');
```

Comment:
I have modified it, when I run the case with `node test/async_hooks/test-promise.promise-before-init-hooks.js`, the `uid` of the parent promise will be `5`.

But when run `make -j4 test`, it will be `2`.

I will try to figure out why.